packages/financial-templates-lib/src/price-feed/HmLandRegistryPriceFeed.ts

```diff
@@ -0,0 +1,253 @@
+    private readonly lookback: number, // lookback should ideally be 90 days
+    private readonly networker: NetworkerInterface,
+    private readonly getTime: () => Promise<number>,
+    private readonly priceFeedDecimals = 18,
+    private readonly minTimeBetweenUpdates = 900 // 15 mins is a reasonable default since this API uses an interval of 15min
+  ) {
+    super();
+
+    this.uuid = `HmLandRegistry-${index}`;
+
+    this.priceHistory = [];
+
+    this.convertPriceFeedDecimals = (number) => {
+      return Web3.utils.toBN(
+        parseFixed(number.toString().substring(0, priceFeedDecimals), priceFeedDecimals).toString());
+    };
+  }
+  public async update(): Promise<void> {
+    const currentTime = await this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== null && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "HmLandRegistryPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTime + this.minTimeBetweenUpdates - currentTime,
+      });
+      return;
+    }
+
+    const TIMEZONE = Intl.DateTimeFormat().resolvedOptions().timeZone;
+    const endDateTime = currentTime;
+    const startLookbackWindow = endDateTime - this.lookback;
+
+    const endDateTimeString = this._secondToDate(currentTime);
+    const startDateTimeString = this._secondToDate(startLookbackWindow);
+    console.log("endDateTimeString", endDateTimeString);
+
+    this.logger.debug({
+      at: "HmLandRegistryPriceFeed",
+      message: "Updating HmLandRegistryPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime,
+      timezone: TIMEZONE,
+    });
+
+    // 1. Construct URL & SPARQL query.
+    
+    const url = "http://landregistry.data.gov.uk/landregistry/query";
+    
+    const sparqlQuery = `
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX sr: <http://data.ordnancesurvey.co.uk/ontology/spatialrelations/>
+PREFIX ukhpi: <http://landregistry.data.gov.uk/def/ukhpi/>
+PREFIX lrppi: <http://landregistry.data.gov.uk/def/ppi/>
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+PREFIX lrcommon: <http://landregistry.data.gov.uk/def/common/>
+
+# House price index for a specific region within a given date range
+SELECT ?region ?date ?hpi
+{
+  ?obs ukhpi:refRegion ?region ;
+       ukhpi:refPeriodStart ?date ;
+       ukhpi:housePriceIndex ?hpi .
+
+  FILTER (
+    ?date > "${startDateTimeString}"^^xsd:date &&
+    ?date <= "${endDateTimeString}"^^xsd:date
+  )
+
+  FILTER (?region = <http://landregistry.data.gov.uk/id/region/united-kingdom>)
+}
+ORDER BY DESC(?date)
```

Comment:
Change this to `ORDER BY ASC(?date)`. Code is returning incorrect current price.

The below matching code works with ASC order data only. See this comment _// historicalPricePeriods are ordered from oldest to newest._